### PR TITLE
feat: add IconPicker

### LINKED ISSUE #1657: [Timebox] Add an IconPicker to twenty ui library
https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=11024-197483&mode=design&t=ItVXoMtBBrLdH8c7-0

Image

- Display all "vetted" react tabler icons within a sublist (similar to what we have in @/ui/icons
- be able to search through the possibilities
- be able to pass a selectedIcon
- get out the new selectedIcon onChange

Make sure that when opened, the icon currently selected is visible (Use the hover state as active state)

### PR DESCRIPTION
**Pull request by thaisguigon**
Closes #1657

### CODE REVIEW

File: front/src/modules/ui/input/components/IconPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+import { LightIconButton } from '@/ui/button/components/LightIconButton';
+import { DropdownMenuSearchInput } from '@/ui/dropdown/components/DropdownMenuSearchInput';
+import { StyledDropdownMenu } from '@/ui/dropdown/components/StyledDropdownMenu';
+import { StyledDropdownMenuItemsContainer } from '@/ui/dropdown/components/StyledDropdownMenuItemsContainer';
+import { StyledDropdownMenuSeparator } from '@/ui/dropdown/components/StyledDropdownMenuSeparator';
+import { IconComponent } from '@/ui/icon/types/IconComponent';
+
+type IconPickerProps = {
+  icons: Record<string, IconComponent>;
```
Comment (charlesBochet) [inline front/src/modules/ui/input/components/IconPicker.tsx:12]: @thaisguigon let's merge it but I think we should not pass the icons to the iconPicker, it should be part of it.
the icon picker should take an "icon" url or an iconList (enum = 'tabler-icons' only for now) and load them asynchronously internally
This can be treated in another PR if you prefer!

File: front/src/modules/ui/input/components/IconPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  onChange: (iconName: string) => void;
+  selectedIconName?: string;
```
Comment (charlesBochet) [inline front/src/modules/ui/input/components/IconPicker.tsx:14]: selectedIconKey
(let's use key and not name that could be confused with a "displayed" name)

**Review by charlesBochet — APPROVED** [review]

File: front/src/modules/ui/input/components/IconPicker.tsx (hunk 2 above)
Comment (thaisguigon) [reply front/src/modules/ui/input/components/IconPicker.tsx]: Done!

File: front/src/modules/ui/input/components/IconPicker.tsx (hunk 1 above)
Comment (thaisguigon) [reply front/src/modules/ui/input/components/IconPicker.tsx:12]: Ok, we might want to pass the icon key + the icon component to `onChange` then.
This is a first, generic implementation, I was thinking we could discuss the icon loading method later and implement it in another PR/issue.

Comment (charlesBochet) [reply front/src/modules/ui/input/components/IconPicker.tsx:12]: ok!

**Review by charlesBochet — APPROVED** [review]
